### PR TITLE
Chore: Refactor Prometheus HTTP transport and replace with HTTP client middleware

### PR DESCRIPTION
**Pull request by marefr**
**What this PR does / why we need it**:
Following #33439 this refactors the Prometheus HTTP transport which is replaced by HTTP client middleware.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
- Removes applying basic auth in transport since that should be applied by the HTTP client provider automatically
- Changes tracing span operation name from alerting.prometheus to datasource.prometheus

### CODE REVIEW

**Review by ivanahuckova — APPROVED** [review]
I have tested this and auth & custom params works as before. Code looks cleaner and is more tested. LGTM! 🚀 🤩

File: pkg/tsdb/prometheus/custom_query_params_middleware.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+			params := url.Values{}
+			for _, param := range strings.Split(customQueryParams, "&") {
+				parts := strings.Split(param, "=")
+				if len(parts) == 1 {
+					// This is probably a mistake on the users part in defining the params but we don't want to crash.
+					params.Add(parts[0], "")
+				} else {
+					params.Add(parts[0], parts[1])
+				}
+			}
```
Comment (zserge) [inline pkg/tsdb/prometheus/custom_query_params_middleware.go:38]: Wouldn't https://golang.org/pkg/net/url/#ParseQuery make code simpler in this case?

**Review by zserge — APPROVED** [review]

File: pkg/tsdb/prometheus/custom_query_params_middleware.go (hunk 1 above)
Comment (marefr) [reply pkg/tsdb/prometheus/custom_query_params_middleware.go:38]: Probably :+1: Not sure I want to change that in this PR though. Perhaps something for Obervability squad to look into or I'll open a separate PR for that.

Comment (marefr) [reply pkg/tsdb/prometheus/custom_query_params_middleware.go:38]: https://github.com/grafana/grafana/pull/34814

**Pull request merged**